.github/workflows/newfold-prepare-release.yml

```diff
@@ -0,0 +1,40 @@
+name: Newfold Prepare Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      level:
+        description: The level of release to be used.
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+        default: "patch"
+        required: true
+
+jobs:
+  # This job is used to extract the branch name from the pull request or branch.
+  setup:
+    name: Setup
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.extract_branch.outputs.branch }}
+    steps:
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_OUTPUT"
+        id: extract_branch
+
+  # This job runs the newfold plugin-prep-release workflow for this plugin.
+  prep-release:
+    name: Prepare Release
+    needs: setup
+    uses: newfold-labs/workflows/.github/workflows/reusable-plugin-prep-release.yml@add/plugin-prep-release
+    with:
+      plugin-repo: ${{ github.repository }}
+      plugin-branch: ${{ needs.setup.outputs.branch }}
+      level: ${{ inputs.level }}
+      json-file: "package.json"
+      php-file: "bluehost-wordpress-plugin.php"
+    secrets: inherit
```

Comment:
What specific secret is required by the reusable workflow? I don't see any `secrets.**` references.

If one is required, we should explicitly specify it in the reusable workflow under `workflow_call` and pass it here to adhere to the principle of least privilege.